include/deal.II/base/config.h.in

```diff
@@ -138,7 +138,7 @@
 #define DEAL_II_COMPILER_VECTORIZATION_LEVEL 0
 #endif
 
-#define DEAL_II_HAVE_ARM_NEON @DEAL_II_HAVE_ARM_NEON@
+#cmakedefine DEAL_II_HAVE_ARM_NEON
```

Comment:
What's the difference in behavior? Did we forget to somehow "export" `DEAL_II_HAVE_ARM_NEON`?

Reply:
We test elsewhere if the macro is defined and that is true even when `DEAL_II_HAVE_ARM_NEON` is `0` (indicating the CMake check failed).

Reply:
@bangerth The line `#define FOO @FOO@` will always expand to
```
#define FOO <something including empty string>
```
In contract `#cmakedefine FOO` only expands the macro if the CMake variable `FOO` is defined and evaluating to true. Meaning, we get `#define FOO`, or `/* #undef FOO */` depending on the state of the variable.

Note that we are using the following preprocessor guard:
`#if defined(DEAL_II_HAVE_ARM_NEON) && defined (__ARM_NEON)`
rendering the `DEAL_II_HAVE_ARM_NEON` conditional in this expression useless.

Reply:
Also, it would probably be cleanest to not export this variable in `config.h`. We already have the `VECTORIZATION_LEVEL_IN_BITS` variable available (that is used for x86):
```
#if DEAL_II_VECTORIZATION_WIDTH_IN_BITS >= 128 && !defined(__ARM_NEON)
#endif
```

I might change this in a follow-up pull request. I have just spun up an arm64 cloud instance to test these changes.

Reply:
Yes, I agree, this would be a good option, that is better and makes the NEON more symmetric to x86.

Reply:
Oh, oh, doh. Thanks for (re)educating me with the difference between an empty `#define`  and `#undef`. I might have my strengths, but the preprocessor ain't one.